libespm/include/libespm/CTDA.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+#include <cstdint>
+
+namespace espm {
+
+// Not a record type, but a field type commonly used across different record
+// types
+struct CTDA
+{
+  enum class Operator : uint8_t
+  {
+    EqualTo = 0,
+    NotEqualTo = 1,
+    GreaterThen = 2,
+    GreaterThenOrEqualTo = 3,
+    LessThen = 4,
+    LessThenOrEqualTo = 5
+  };
+
+  enum class Flags : uint8_t
+  {
+    ANDORDEFAULT = 0x00,
+    OR = 0x01,
+    Parameters = 0x02,
+    UseGlobal = 0x04,
+    UsePackData = 0x08,
+    SwapSubject = 0x10
+  };
+
+  enum class RunOnTypeFlags : uint32_t
+  {
+    Subject = 0,
+    Target = 1,
+    Reference = 2,
+    CombatTarget = 3,
+    LinkedReference = 4,
+    QuestAlias = 5,
+    PackageData = 6,
+    EventData = 7
+  };
+
+  struct DefaultData
+  {
+    uint32_t firstParameter;
+    uint32_t secondParameter;
+  };
+  static_assert(sizeof(DefaultData) == 8);
+
+  struct EventData
+  {
+    uint16_t eventFunction;
+    char eventMember[2];
+    uint32_t parameter;
+  };
+  static_assert(sizeof(EventData) == 8);
+
+  uint8_t operatorFlag;
+  uint8_t unknown[3];
+  float comparisonValue;
+  uint16_t functionIndex;
+  uint8_t padding[2];
+
+  // Default/Event data
+  uint8_t functionData[8];
+
+  RunOnTypeFlags runOnType;
+  int32_t reference;
+  int32_t unknown2;
+
+  Operator GetOperator()
+  {
+    uint8_t firstBit = (uint8_t)((operatorFlag & 0x80) ? 4 : 0);
```

Comment:
Please use static_cast instead of C-style cast

example:

static_cast<uint8_t>((operatorFlag & 0x80) ? 4 : 0)